src/sql/workbench/browser/modelComponents/declarativeTable.component.ts

```diff
@@ -110,6 +118,13 @@ export default class DeclarativeTableComponent extends ComponentBase implements
 		this.onCellDataChanged(e, rowIdx, colIdx);
 	}
 
+	public onHeaderCheckBoxChanged(e: boolean, colIdx: number): void {
+		this.data.forEach((row, rowidx) => {
+			this.onCellDataChanged(e, rowidx, colIdx);
```

Comment:
We should only call this if the actual data changed otherwise we'll be sending out extra events we may not need to. 

Although given the method as it is currently you could also just make it so if the value is the same as the current one then it just returns immediately.